src/pybind/mgr/dashboard/frontend/src/app/shared/components/usage-bar/usage-bar.component.ts

```diff
@@ -6,20 +6,29 @@ import { Component, Input, OnChanges } from '@angular/core';
   styleUrls: ['./usage-bar.component.scss']
 })
 export class UsageBarComponent implements OnChanges {
+  @Input()
+  totalAmount: number;
+  @Input()
+  usedAmount: number;
```

Comment:
I'm not sure this is needed. It would be easier to just add a flag like 
```suggestion
  @Input()
  isBinary = true;
```
Than rename `totalBytes` into `total` and `usedBytes` into `used`. 

Resulting in a cleaner template:

For used:
```html
      <td class="text-right"><strong> {{ used | isBinary ? dimlessBinary : dimless }}</strong></td>
```

For free:
```html
      <td class="text-right"><strong> {{ used - total | isBinary ? dimlessBinary : dimless }}</strong></td>
```

Reply:
The reason behind having extra 2 input variables is we need some variable as a comparison between size and object usage in terms of bytes and raw number. To make it more clear, in bucket-list component html , size template only passes total/used bytes as input so total/used amount is 0 and vice-versa in case of object template. So we can use totalAmount as a common variable for both templates in order to know which type of data we need to show , which is bytes in case of size usage and raw numbers in case of objects usage. 

But if don't use these 2 extra variables then we will not be able to distinguish this difference , as total will always be greater than 0

Reply:
If we use the boolean `isBinary` it should be pretty clear. I can't come up with a right case in which we would need both at the same time.

Reply:
Maybe we don't need usedAmount variable , but we do need totalAmount. I will try explaining through a example. Suppose if I set quota for bucket as max_objects=10 and max_size=1024B, so in this case if we just use 2 variables(total, used) total > 0 for both size and object template (in bucket-list component html). So data will be shown in the same format , either in bytes or just the number for both size and object usage. But what we need to shown is bytes for size and raw numbers for objects. So that's the reason we need a extra variable for comparison

Reply:
I'm not sure where the miss understanding took place. I'll try to explain it again:

You have two bar usages. One uses bytes (size), the other raw numbers (objects). Once they are shown, each bar represents it's own `cd-usage-bar` instance, this way they are not sharing the same variable states. That said the size instance can hold 1024B in the variable called `total` and the objects instance hold 10 in the variable called `total` at the same time because it's not the same instance.

If they share some state it's a bug in `cd-usage-bar` and it would be nice if you can fix it.

Reply:
Yes, I got that. But as per the example we have total > 0 , so if we have some condition on total for e.g. if we set the condition to show data in bytes in case total > 0 then it will show data in bytes for object too

Reply:
Sorry, my mistake here, I didn't see isBinary as input variable. Now got it , thanks!